awx/main/scheduler/task_manager.py

```diff
@@ -250,6 +254,11 @@ def start_task(self, task, rampart_group, dependent_tasks=None, instance=None):
                 task.controller_node = controller_node
                 logger.debug('Submitting isolated {} to queue {} controlled by {}.'.format(
                              task.log_format, task.execution_node, controller_node))
+            elif container_controller:
+                task.instance_group = rampart_group
+                pod_manager = PodManager(task)
+                pod_manager.deploy_pod()
+                task.execution_node = pod_manager.pod_name
```

Comment:
`execution_node` is the container and the `instance_group`  is the first of the `preferred_instance_groups` which comes from our existing hierarchy of which instance group to choose logic :+1:

Reply:
It's a shame we don't capture the controller node.

Reply:
We don't?

https://github.com/ansible/awx/blob/5d6916f69ee1ce39676384514fc0cb79cbe3bdaa/awx/main/models/unified_jobs.py#L586-L590